src/gui/application.cpp

```diff
@@ -551,51 +544,105 @@ namespace Toolbox {
         }
     }
 
-    void FileDialog::OpenDialog(std::filesystem::path starting_path,
-                                GLFWwindow* parent_window,
-                                bool is_directory,
-                                std::optional<std::vector<std::pair<
-                                  const char*, const char*>>> maybe_filters) {
+#ifdef TOOLBOX_PLATFORM_WINDOWS
+    void FileDialog::OpenDialog(
+        std::filesystem::path starting_path, GLFWwindow *parent_window, bool is_directory,
+        std::optional<std::vector<std::pair<std::string, std::string>>> maybe_filters) {
```

Comment:
Seems like most of the code in these two platform branches is duplicated between branches, since it's a cross-platform api. I try to avoid duplicated code because it's easy for bugs to sneak in when you accidentally change something in one copy but not the other. If we switch to using NFD_PickFolderU8_With and NFD_OpenDialogU8_With then we can pass char*'s on both platforms and only have one copy of this code. Alternatively, we could just move the platform branching to inside the function in a few key places, so most of the code is shared between platforms.